include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -3509,6 +3511,317 @@ namespace internal
 
       return mapping_info;
     }
+
+    /**
+     * This function provides information which L2 DoF index is associated with
+     * a support point. Support points are numberd according to H1 DoFs.
+     *
+     * @param[in] dof_handler_l2 L2 conforming DoFHandler
+     * @param[in] dof_handler_h1 H1 conforming DoFHandler
+     * @param[in] constraint AffineConstrains associated with @p dof_handler_l2. Only unconstrained DoFs are considered
+     * @return a pair of vectors which resembles a CRS structure to access global L2 DoF indices from local support point (or H1 DoF) indices.
+     */
+    template <int dim, int spacedim, typename Number>
+    std::pair<std::vector<unsigned int>, std::vector<types::global_dof_index>>
+    support_points_to_l2_dofs(
+      const DoFHandler<dim, spacedim> &        dof_handler_l2,
+      const DoFHandler<dim, spacedim> &        dof_handler_h1,
+      const dealii::AffineConstraints<Number> &constraint)
+    {
+      Assert(dof_handler_l2.get_fe().conforming_space ==
+               FiniteElementData<dim>::Conformity::L2,
+             ExcMessage("dof_handler_l2 not L2 conforming."));
+      Assert(dof_handler_h1.get_fe().conforming_space ==
+               FiniteElementData<dim>::Conformity::H1,
+             ExcMessage("dof_handler_h1 not H1 conforming."));
+
+      Assert(dof_handler_h1.get_fe().degree == dof_handler_l2.get_fe().degree,
+             ExcMessage("DoFhandlers need the same degree."));
+      Assert(dof_handler_h1.get_fe().n_components() ==
+               dof_handler_l2.get_fe().n_components(),
+             ExcMessage("DoFhandlers need the same number of components."));
+      Assert(&dof_handler_l2.get_triangulation() ==
+               &dof_handler_l2.get_triangulation(),
+             ExcMessage("DoFhandlers need the same underlying triangulation."));
+
+      const auto &tria          = dof_handler_l2.get_triangulation();
+      const auto  degree        = dof_handler_l2.get_fe().degree;
+      const auto  dofs_per_cell = dof_handler_l2.get_fe().n_dofs_per_cell();
+
+      const Utilities::MPI::Partitioner partitioner_h1(
+        dof_handler_h1.locally_owned_dofs(),
+        DoFTools::extract_locally_active_dofs(dof_handler_h1),
+        dof_handler_h1.get_communicator());
+
+      const Utilities::MPI::Partitioner partitioner_l2(
+        dof_handler_l2.locally_owned_dofs(),
+        DoFTools::extract_locally_relevant_dofs(dof_handler_l2),
+        dof_handler_l2.get_communicator());
```

Comment:
```suggestion
```

Reply:
Done [f85dd34](https://github.com/dealii/dealii/pull/15209/commits/f85dd34bc167f7f13689909da7fda3bb07dd2ec4)

Reply:
After refactoring the internal functions, I had to restore the partitioner to be able to translate global to local index positions, which are needed to check if a DoF has been processed by accessing a vector of bools. See [2b815bf](https://github.com/dealii/dealii/pull/15209/commits/2b815bf4b670ea2a8ca42852d1299df5a918a95b).